### PR TITLE
Use the telnet output to know when the debug protocol port is open

### PR DESCRIPTION
**Pull request by Christian-Holbrook**

### CODE REVIEW

File: src/adapters/DebugProtocolAdapter.ts [hunk 1]
```diff
@@ -313,7 +313,19 @@ export class DebugProtocolAdapter {
                 this.emit('diagnostics', diagnostics);
             });
 
+            this.socketDebugger.on('control-connected', () => {
+                this.connectionDeferred.resolve();
+
+                this.logger.log(`Closing telnet connection used for compile errors`);
```
Comment (TwitchBronBron) [inline src/adapters/DebugProtocolAdapter.ts:319]: Let's not close this. The compileClient (telnet connection) should stay alive for the lifetime of the debug session (i.e. until someone clicks "stop")

File: src/adapters/DebugProtocolAdapter.ts [hunk 2]
```diff
@@ -44,6 +44,12 @@ export class DebugProtocolAdapter {
         });
     }
 
+    private connectionDeferred = defer<void>();
+
+    public deferredConnectionPromise(): Promise<void> {
```
Comment (TwitchBronBron) [inline src/adapters/DebugProtocolAdapter.ts:49]: Maybe call it something more like this? The word 'deferred' isn't necessary here, that's an internal implementation detail
```suggestion
    public isConnected(): Promise<void> {
```

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Looking good! Just a few comments

File: src/adapters/DebugProtocolAdapter.ts (hunk 1 above)
Comment (Christian-Holbrook) [reply src/adapters/DebugProtocolAdapter.ts:319]: This was a small step to start using the telnet logs to trigger the control port connection.

I'm starting the work to keep the debug session open once the control port closes.

**Pull request closed without merging**